azure-pipelines.yml

```diff
@@ -69,12 +69,35 @@ jobs:
         python.version: '3.7'
       py38:
         python.version: '3.8'
+  variables:
+    PIP_CACHE_DIR: $(Pipeline.Workspace)/.pip
+    CCACHE_DIR: $(Pipeline.Workspace)/ccache
   steps:
   - task: UsePythonVersion@0
     inputs:
       versionSpec: '$(python.version)'
     displayName: 'Use Python $(python.version)'
 
+  - script: |
+      sudo apt update && sudo apt install -y ccache
+      echo "##vso[task.prependpath]/usr/lib/ccache"
+    displayName: Install ccache and update PATH to use linked versions of gcc, cc, etc
+
+  - task: Cache@2
+    inputs:
+      key: 'ccache | "$(Agent.OS)" | "$(python.version)"'
+      path: $(CCACHE_DIR)
+    displayName: Cache C compilation with ccache
+
+  - task: Cache@2
+    inputs:
+      key: 'python | "$(Agent.OS)"'
+      restoreKeys: |
```

Comment:
this is more out of curiosity than a review, but why do we need `restoreKeys` for the pip cache and not for the ccache?

Reply:
I didn't give this much thought.
I just copied it from their documentation.